addons/block_code/ui/picker/categories/category_factory.gd

```diff
@@ -192,3 +192,93 @@ static func add_to_categories(main: Array[BlockCategory], addition: Array[BlockC
 			main.append(add_category)
 
 	return main
+
+
+static func built_in_type_to_block_type(type: Variant.Type):
+	match type:
+		TYPE_BOOL:
+			return Types.BlockType.BOOL
+		TYPE_INT:
+			return Types.BlockType.INT
+		TYPE_FLOAT:
+			return Types.BlockType.FLOAT
+		TYPE_STRING:
+			return Types.BlockType.STRING
+		TYPE_VECTOR2:
+			return Types.BlockType.VECTOR2
+
+	return null
+
+
+static func property_to_blocklist(property: Dictionary) -> Array[Block]:
```

Comment:
Nice! I love this factory function. It can also have a "Change property by [amount]" that translates to `property += amount` as a sugar syntax to "Set property to [value]" and a "a + b" block. See MS Arcade for reference.

Reply:
This is a great idea and I will add the simple version of it now that adds two of the same type together. I think it will be even better once we are able to add a float to a vector and so on.